os/drivers/mipidsi/mipi_dsi_packet.c

```diff
@@ -0,0 +1,235 @@
+ *   otherwise.
+ *
+ ****************************************************************************/
+
+bool mipi_dsi_packet_format_is_short(uint8_t type)
+{
+	switch (type) {
+	case MIPI_DSI_VSYNC_START:
+	case MIPI_DSI_VSYNC_END:
+	case MIPI_DSI_HSYNC_START:
+	case MIPI_DSI_HSYNC_END:
+	case MIPI_DSI_COMPRESSION_MODE:
+	case MIPI_DSI_END_OF_TRANSMISSION:
+	case MIPI_DSI_COLOR_MODE_OFF:
+	case MIPI_DSI_COLOR_MODE_ON:
+	case MIPI_DSI_SHUTDOWN_PERIPHERAL:
+	case MIPI_DSI_TURN_ON_PERIPHERAL:
+	case MIPI_DSI_GENERIC_SHORT_WRITE_0_PARAM:
+	case MIPI_DSI_GENERIC_SHORT_WRITE_1_PARAM:
+	case MIPI_DSI_GENERIC_SHORT_WRITE_2_PARAM:
+	case MIPI_DSI_GENERIC_READ_0_PARAM:
+	case MIPI_DSI_GENERIC_READ_1_PARAM:
+	case MIPI_DSI_GENERIC_READ_2_PARAM:
+	case MIPI_DSI_DCS_SHORT_WRITE_0_PARAM:
+	case MIPI_DSI_DCS_SHORT_WRITE_1_PARAM:
+	case MIPI_DSI_DCS_READ_0_PARAM:
+	case MIPI_DSI_EXECUTE_QUEUE:
+	case MIPI_DSI_SET_MAXIMUM_RETURN_PACKET_SIZE:
+		return true;
+	}
+
+	return false;
+}
+
+/****************************************************************************
+ * Name: mipi_dsi_packet_format_is_long
+ *
+ * Description:
+ *   Check if a packet is of the long format
+ *
+ * Input Parameters:
+ *   type - MIPI DSI data type of the packet
+ *
+ * Returned Value:
+ *   True if the packet for the given data type is a long packet, false
+ *   otherwise.
+ *
+ ****************************************************************************/
+
+bool mipi_dsi_packet_format_is_long(uint8_t type)
+{
+	switch (type) {
+	case MIPI_DSI_NULL_PACKET:
+	case MIPI_DSI_BLANKING_PACKET:
+	case MIPI_DSI_LONG_GENERIC_WRITE:
+	case MIPI_DSI_DCS_LONG_WRITE:
+	case MIPI_DSI_PICTURE_PARAMETER_SET:
+	case MIPI_DSI_COMPRESSED_PIXEL_STREAM:
+	case MIPI_DSI_LOOSELY_PACKED_PIXEL_STREAM_YCBCR20:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_YCBCR24:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_YCBCR16:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_RGB30:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_RGB36:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_YCBCR12:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_RGB16:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_RGB18:
+	case MIPI_DSI_PIXEL_STREAM_3BYTE_RGB18:
+	case MIPI_DSI_PACKED_PIXEL_STREAM_RGB24:
+		return true;
+	}
+
+	return false;
+}
+
+/****************************************************************************
+ * Name: mipi_dsi_create_packet
+ *
+ * Description:
+ *   Create a packet from a message according to the DSI protocol
+ *
+ * Input Parameters:
+ *   packet - Pointer to a DSI packet structure
+ *   msg    - Message to translate into a packet
+ *
+ * Returned Value:
+ *   Return: 0 on success or a negative error code on failure.
+ *
+ ****************************************************************************/
+
+int mipi_dsi_create_packet(FAR struct mipi_dsi_packet *packet, FAR const struct mipi_dsi_msg *msg)
+{
+	if (packet == NULL) {
+		printf("packet is NULL \n");
+		return -EINVAL;
+	}
+	if (msg == NULL) {
+		printf("msg is NULL \n");
```

Comment:
MIPI seems to be an individual driver which is tightly coupled with LCD, maybe you want to define another logging type for it. etc: `mipidbg`, `mipivdbg`? Please find os/include/debug.h.
I'm not very sure, maybe @sunghan-chang could provide better comment on this.